src/components/tables/FieldSelection/index.tsx

```diff
@@ -76,6 +81,20 @@ function FieldSelectionTable({ projections }: Props) {
         },
     };
 
+    const searchQuery = useBinding_searchQuery();
+
+    const processedProjections = useMemo(
+        () =>
+            searchQuery
+                ? projections?.filter(
+                      ({ field, ptr }) =>
+                          field.includes(searchQuery) ||
+                          ptr?.includes(searchQuery)
+                  )
+                : projections,
+        [projections, searchQuery]
+    );
```

Comment:
This is good for now - might want to look into making this a shallow hook so we can keep the rendering scope a bit smaller.